server/controllers/TripController.js

```diff
@@ -304,6 +304,40 @@ class TripController {
       throw new Error('Could not delete trip')
     }
   }
+
+  async enableShareTrip(userId, id) {
+    try {
+      await TripModel.updateOne(
+        { _id: id, userId: userId },
+        {
+          $set: {
+            isPublic: true, // update the field
+          },
+        },
+        function (err) {
+          if (err) throw err
+          console.log('1 document updated')
+        },
+      ).lean()
+    } catch (error) {
+      if (config.server.env === 'DEV')
```

Comment:
No more of these if(config.server.env === 'DEV') things, see just throw the error, and in the route, call next(err) :')